# NULL HYPOTHESIS:

##  The commute time is the same or longer with the new bus route as it was before.

### H0: Average_Time_new >= Average_Time_old


# Alternative  HYPOTHESIS:

## the commute time is shorter with the new bus route as it was before.

###  Ha: Average_Time_new < Average_Time_old


## I will use a significance level  $\alpha=0.05$


## z =  $\frac{μ_{old} -  μ_{new}}{σ_{old} / √N}$


In [91]:
import numpy as np
import requests
import math

# the download of the data
t = requests.get('https://raw.github.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt')
t_l = t.text.split('\n')
time=[]
for i in range(len(t_l)-1):
    #print(t_l[i])
    n=(float(t_l[i]))
    time.append(n)
times= np.array(time)   
Mean_new = times.mean()
Std_new = times.std()
Mean_old=36
Std_old=6
N= times.size
print("μ of the new bus route time is %f" % (Mean_new))

μ of the new bus route is 34.466162


In [93]:
z_statistics= (Mean_old - Mean_new) / (Std_old / math.sqrt(N))
print("Z Statistics is %f" % (z_statistics))

Z Statistics is 2.556397



### μ_old = 36, μ_new = 34.466162
### σ_old = 6, N =100

## z =  $\frac{μ_{old} -  μ_{new}}{σ_{old} / √N}$ = $\frac{36 -  34.466162}{6 / √100}$  = 2.556397

### As the Z test is returning multiples of the standard deviation of a Normal distribution,  a Z Statistics  of 2 represents 2 standard deviation away from the mean, which corresponde to 95% of the data and the 0.05 significance level. 
### Since the Z Statistics now is 2.556 greater than 2,  the sample mean(new route time)  is more than 2  standard deviation away from the population mean(old route time) , which means it is out of the 95% of data and greater than the 0.05 significance level, so it fails to reject the null hypothesis that the commute time is the same or longer with the new bus route as it was before. 